llvm/tools/sycl-post-link/CompileTimePropertiesPass.cpp

```diff
@@ -142,9 +153,129 @@ PreservedAnalyses CompileTimePropertiesPass::run(Module &M,
     }
   }
 
+  // Check pointer annotations.
+  SmallVector<IntrinsicInst *, 4> RemovableAnnots;
+  for (Function &F : M)
+    for (BasicBlock &BB : F)
+      for (Instruction &Inst : BB)
```

Comment:
Nit: you can use instruction iterator https://llvm.org/docs/ProgrammersManual.html#iterating-over-the-instruction-in-a-function

Reply:
I have switched to using this. Thanks!